qiskit/opflow/primitive_ops/pauli_op.py

```diff
@@ -321,11 +321,12 @@ def to_circuit(self) -> QuantumCircuit:
             qc.global_phase = -phase * pi / 2
             return qc
 
-        if self.num_qubits == 1:
-            gate = {"I": IGate(), "X": XGate(), "Y": YGate(), "Z": ZGate()}[pauli]
+        if self.num_qubits == 1 and pauli != "I":
+            gate = {"X": XGate(), "Y": YGate(), "Z": ZGate()}[pauli]
+            qc.append(gate, [0])
         else:
             gate = PauliGate(pauli)
```

Comment:
This pushes the `pauli == "I"` case into the `else` branch, whereas it didn't previously.  I think it should end up in the correct behaviour occurring eventually, but it's going to add an unnecessary intermediate gate into the definition, when there's already logic in place to skip `I`.

(Also, this is probably just me being overly picky with a manual memory-management hat, but it seems slightly wasteful to create 4 intermediate objects here - the dictionary, and three gate instances.  The form in the circuit library which has the dict of constructors and then does `gate()` requires half as many temporary objects.)

Reply:
Oh yeah, it definitely wasn't supposed to hit the else-case! Fixed in 1ae6e9582 (along with the instance creation 🙂)

Reply:
It'd be good to stick in a small test that explicitly hits this - I think just testing that `PauliOp("I").to_circuit()` has the right qubits and clbits and no data would do?

Reply:
Added in 759944e